doc/src/build/programming-with-objects/ch4-object-wrapping.md

```diff
@@ -0,0 +1,195 @@
+
+First of all, let's define such an object type:
+```rust
+struct Object has key, store {
+    id: VersionedID,
+    scarcity: u8,
+    style: u8,
+}
+```
+In a real application, we probably would make sure that there is a limited supply of the objects and there is a mechanism to mint them to a list of owners. For simplicity and demonstration purpose, here we will just make it straightforward to create:
+```rust
+public(script) fun create_object(scarcity: u8, style: u8, ctx: &mut TxContext) {
+    let object = Object {
+        id: TxContext::new_id(ctx),
+        scarcity,
+        style,
+    };
+    Transfer::transfer(object, TxContext::sender(ctx))
+}
+```
+Anyone can call `create_object` to create a new object with specified `scarcity` and `style`. The created object will be sent to the signer of the transaction. We will likely also want to be able to transfer the object to others:
+```rust
+public(script) fun transfer_object(object: Object, ctx: &mut TxContext) {
+    Transfer::transfer(object, TxContext::sender(ctx))
+}
+```
+
+Now let's look at how we could enable a swap/trade between your object and others' objects. A straightforward idea is this: define a function that takes two objects from two accounts, and swap their ownership. But this doesn't work in Sui! Recall from Chapter 2 that only object owners can send a transaction to mutate the object. So one person cannot send a transaction that would swap their own object with someone else's object.
+
+In the future, we will likely introduce multi-sig transactions so that two people can sign the same transaction for this type of usecase. However you may not always be able to find someone to swap with right away. A multi-sig transaction won't work in this scenario. Even if you can, you may not want to carry the burden of finding a swap target.
+
+Another common solution is to "send" your object to a pool (e.g. a marketplace in the case of NFT, or a liquidity pool in the case of tokens), and perform the swap in the pool (either right away, or latter when there is demand). In future chapters, we will explore the concept of shared objects that can be mutated by anyone, and show that how it enables anyone to operate in a shared object pool. In this chapter, we will focus on how to achieve the same effect using owned objects. Transactions only using owned objects is significantly faster and cheapter (in terms of gas) than using shared objects, since it does not require consensus in Sui.
```

Comment:
```suggestion
Another common solution is to "send" your object to a pool (e.g. a marketplace in the case of NFT, or a liquidity pool in the case of tokens), and perform the swap in the pool (either right away, or later when there is demand). In future chapters, we will explore the concept of shared objects that can be mutated by anyone, and show that how it enables anyone to operate in a shared object pool. In this chapter, we will focus on how to achieve the same effect using owned objects. Transactions only using owned objects is faster and cheaper (in terms of gas) than using shared objects, since it does not require consensus in Sui.
```